metagraph/src/graph/alignment/dbg_aligner.hpp

```diff
@@ -393,7 +401,7 @@ ::align_aggregate(DBGQueryAlignment &paths,
     );
 
     path_queue.call_alignments([&](auto&& alignment) {
-        assert(alignment.is_valid(graph_, &config_));
+        assert(alignment.is_valid(get_graph(), &get_config()));
```

Comment:
why get_graph()? just call graph_ directly. Same for config_.

Reply:
This method was moved to the abstract `SeedAndExtendAligner` class, which doesn't store a graph or a config.

Reply:
but it does store a reference to it, right?
Or you can always make it so if you can implement these getters.

Reply:
when I said "doesn't store a graph or a config" I meant that it doesn't store pointers, it just as `get_graph` and `get_config` as pure virtual methods

Reply:
Then its children must store them, so you can move those pointers/references to this class.
What is the point to declare these virtual getters?

Reply:
I don't know why it's not showing up for me in "Files changed", but the point is that with this structure, I can now move most of the alignment code out of the header, since most of the alignment functions don't depend `DBGAligner`'s three template classes.

Reply:
Though I could move the references to `ISeedAndExtendAligner` and make them protected, then anything that inherits from it can use them.